arch/arm/src/stm32/hardware/stm32g47xxx_uart.h

```diff
@@ -0,0 +1,420 @@
+#define USART_TDR_MASK                 (0x1ff << USART_TDR_SHIFT)
+#  define USART_TDR(n)                 (((n) << USART_TDR_SHIFT) & USART_TDR_MASK)
+
+/* Prescaler Register */
+
+#define USART_PRESC_PRESCALER_SHIFT    (0)
+#define USART_PRESC_PRESCALER_MASK     (0xf << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_1        (0x0 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_2        (0x1 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_4        (0x2 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_6        (0x3 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_8        (0x4 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_10       (0x5 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_12       (0x6 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_16       (0x7 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_32       (0x8 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_64       (0x9 << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_128      (0xa << USART_PRESC_PRESCALER_SHIFT)
+#define USART_PRESC_PRESCALER_256      (0xb << USART_PRESC_PRESCALER_SHIFT)
+
+/* Compatibility definitions ************************************************/
+
+/* Compatibility with F1/F2/F4 Status Register names */
+
+#define STM32_USART_SR_OFFSET          STM32_USART_ISR_OFFSET
+
+#define USART_SR_PE                    USART_ISR_PE   /* Parity Error */
+#define USART_SR_FE                    USART_ISR_FE   /* Framing error */
+#define USART_SR_NE                    USART_ISR_NE   /* Noise detected flag */
+#define USART_SR_ORE                   USART_ISR_ORE  /* Overrun error */
+#define USART_SR_IDLE                  USART_ISR_IDLE /* IDLE line detected */
+#define USART_SR_RXNE                  USART_ISR_RXNE /* Read Data Register Not Empty */
+#define USART_SR_TC                    USART_ISR_TC   /* Transmission Complete */
+#define USART_SR_TXE                   USART_ISR_TXE  /* Transmit Data Register Empty */
+#define USART_SR_LBD                   USART_ISR_LBDF /* LIN Break Detection Flag */
+#define USART_SR_CTS                   USART_ISR_CTS  /* CTS Flag */
+
+#define USART_SR_ALLBITS               USART_ISR_ALLBITS
+
+#define USART_CR1_M                    USART_CR1_M0
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+#endif /* __ARCH_ARM_SRC_STM32_HARDWARE_STM32G47XXX_UART_H */
```

Comment:
Wow Nathan! - you did a  a lot of work. Stopping here for now, will do more later.